subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/AbstractIncompleteSnapshotWithChildrenTest.groovy

```diff
@@ -0,0 +1,417 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.snapshot
+
+import org.gradle.internal.file.FileType
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import javax.annotation.Nullable
+import java.util.stream.Collectors
+
+@Unroll
+abstract class AbstractIncompleteSnapshotWithChildrenTest<T extends FileSystemNode> extends Specification {
```

Comment:
I think we need to explain what "common prefix" means in this context. Also what "selected child" means.